### ISSUE TITLE
dynamic index pattern and rollover_alias for template_file

### ISSUE THREAD

**Issue by chikinchoi** [uken/fluent-plugin-elasticsearch#771]
#### Problem
Hi, I have an external Fluentd which is connected to Elasticsearch. There are a lot of applications that send logs to the Fluentd.
The different applications will create different indexes and append the %Y.%m to the index name by using "index_name ${taskDef}-%Y.%m". (The taskDef is the application name)
After the log sends to Elasticsearch, we will run below commands in the Kibana dev tool to create the template and rollover index.
```
PUT _template/example_index_name
{
  "index_patterns": ["example_index_name-2020.06*"],                 
    "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 0,
    "index.lifecycle.name": "index-policy",      
    "index.lifecycle.rollover_alias": "example_index_name-2020.06"    
  }
}

PUT example_index_name-2020.06-000001
{
  "aliases": {
    "example_index_name-2020.06": {
      "is_write_index": true
    }
  }
}
```

Now we would like to use the field "template_file" to create the "example_index_name" as the above command. However, may I know how to pass the 'taskDef' field to the template JSON file? 
 

#### Using Fluentd and ES plugin versions

* Fluentd fv1.4-2

**Comment by cosmo0920** [comment]
Could you share your configuration?

**Comment by chikinchoi** [comment]
@cosmo0920 Sure! Below is my fluentd config now and I commanded the template_file field as it didn't work now

fluentd.conf:

<source>
  @type  forward
  @id    input1
  port  24224
  bind 0.0.0.0
</source>

<source>
  @type http
  port 8888
  bind 0.0.0.0
</source>

<filter **firelens**>
  @type concat
  key log
  multiline_start_regexp '^\{\\"@timestamp'
  multiline_end_regexp '/\}/'
  separator ""
  flush_interval 1
  timeout_label @NORMAL
</filter>

<filter **firelens**>
  @type parser
  key_name log
  reserve_data true
  <parse>
  @type json
  </parse>
</filter>

<filter **firelens**>
  @type record_transformer
  enable_ruby
  <record>
    taskDef ${record["ecs_task_definition"].gsub(/:.*/, '')}
  </record>
</filter>

<match **firelens**>
  @type copy
  <store>
    @type elasticsearch
    scheme https
    ssl_version TLSv1_2
    host  "#{ENV['ELASTIC_HOST']}"
    port  "#{ENV['ELASTIC_POST']}"
    user  "#{ENV['ELASTIC_USER']}"
    password "#{ENV['ELASTIC_PWD']}"
    ssl_verify false
    log_es_400_reason true
    index_name ${taskDef}-%Y.%m
# template_file template.json
    <buffer tag, time, ecs_task_definition, taskDef>
      flush_interval 1
      chunk_limit_size 1m
      flush_thread_interval 0.1
      flush_thread_burst_interval 0.01
      flush_thread_count 8
      total_limit_size 1GB
      timekey 1s
      timekey_wait 1s
    </buffer>
  </store>
  <store>
    @type prometheus
    <metric>
      name fluentd_output_status_num_records_total
      type counter
      desc The total number of outgoing records
      <labels>
        tag ${tag}
        hostname ${hostname}
      </labels>
    </metric>
  </store>
  <store>
    @type stdout
  </store>
</match>


template.json:
{
    "index_patterns": ["*"],       #<<I would like to get ${taskDef}-%Y.%m here
      "settings": {
        "number_of_shards": 1,
        "number_of_replicas": 0,
        "index.lifecycle.name": "corpsysmon2-index-policy"     
    }
  }

**Comment by cosmo0920** [comment]
Thanks. I'll investigate about this later.

**Comment by cosmo0920** [comment]
This similar issue in #761. And also fixed in #779.
